File: mobile/lib/features/authentication/domain/repositories/auth_repository.dart
import '../entities/user.dart';

abstract class AuthRepository {
  Future<User?> getCurrentUser();
  Future<void> login(String email, String password);
  Future<void> logout();
  Future<bool> isAuthenticated();
  Future<String?> getAccessToken();
  Future<String?> getRefreshToken();
}

File: app/src/infrastructure/http/auth.repository.js
import { api } from '../api'

export const AuthRepository = {
  login:  (username, password) => api.post('/auth/login/', { username, password }).then((r) => r.data),
  logout: (refresh)            => api.post('/auth/logout/', { refresh }).then(() => {}),
  me:     ()                   => api.get('/auth/me/').then((r) => r.data),
}
